src/game_classification.cpp

```diff
@@ -91,3 +98,65 @@ bool game_classification::is_normal_mp_game() const
 {
 	return this->campaign_type == CAMPAIGN_TYPE::MULTIPLAYER && this->campaign.empty();
 }
+
+namespace {
+
+// helper objects for saved_game::expand_mp_events()
+struct modevents_entry
+{
+	modevents_entry(const std::string& _type, const std::string& _id)
+		: type(_type)
+		, id(_id)
+	{
+	}
+
+	std::string type;
+	std::string id;
+};
+}
+
+std::set<std::string> game_classification::active_addons(const std::string& scenario_id) const
+{
+	//FIXME: this doesn include modsthe current scenario.
+	std::list<modevents_entry> mods;
+	std::set<std::string> loaded_resources;
+	std::set<std::string> res;
+
+	std::transform(active_mods.begin(), active_mods.end(), std::back_inserter(mods),
+		[](const std::string& id) { return modevents_entry("modification", id); }
+	);
+
+	// We don't want the error message below if there is no era (= if this is a sp game).
+	if(!era_id.empty()) {
+		mods.emplace_back(get_tagname(), scenario_id);
+	}
+
+	if(!era_id.empty()) {
+		mods.emplace_back("era", era_id);
+	}
+
+	if(!campaign.empty()) {
+		mods.emplace_back("campaign", campaign);
+	}
+	while(!mods.empty()) {
+
+		const modevents_entry& current = mods.front();
+		if(current.type == "resource") {
+			if(loaded_resources.insert(current.id).second) {
```

Comment:
This seems buggy, if it's meant to do what I think it's meant to do. I think it's meant to skip the rest of the loop for things that have previously been inserted in to loaded_resources; but instead it skips rest of the loop *unless* the id is already in loaded_resources.

Reply:
hmm yes you are right